africanus/calibration/tests/test_utils.py

```diff
@@ -97,7 +110,64 @@ def make_dual_pol_data(sigma_n, n_dir, sigma_f):
     return data_dict
 
 
-def test_residual_vis():
+@corr_shape_parametrization
+def test_corrupt_vis(corr_shape, jones_shape):
+    """
+    Tests corrupt vis against predict_vis. They should do
+    the same thing but corrupt_vis adheres to the structure
+    in the africanus.calibration module.
+    """
+    np.random.seed(42)
+    # simulate noise free data with random DDE's
+    n_dir = 3
+    n_time = 32
+    n_chan = 16
+    n_ant = 7
+    sigma_n = 0.0
+    sigma_f = 0.05
```

Comment:
How about making  some or all of these parametrizations? It would provide the ability to easily try out multiple vales of sigma, for example (0.10 below)

```python
@pytest.mark.parametrize("n_dir", [3])
@pytest.mark.parametrize("sigma_n", [0.0])
@pytest.mark.parametrize("sigma_f", [0.05, 0.10])
def test_corrupt_vis(n_dir, sigma_n, sigma_f, ...):
   data_dict = calibration_factory(sigma_n, sigma_f, ...)
```

Reply:
I will do this if we need more thorough testing of one of the parameters. I am happy with the tests as they are for now